implementations/rust/ockam/ockam_api/src/nodes/service.rs

```diff
@@ -67,8 +69,9 @@ pub struct NodeManager {
 
     skip_defaults: bool,
 
-    pub(crate) vault: Option<Vault>,
+    vault: Option<Vault>,
     identity: Option<Identity<Vault>>,
+    authorities: Option<Vec<PublicIdentity>>,
```

Comment:
```suggestion
    authorities: Vec<PublicIdentity>,
```

Reply:
Could you please elaborate why you think this should changed?

Reply:
There is no need for an `Option` on top of a `Vec` when the `Vec` already has a neutral element. If the vector is empty there are no authorities configured. Adding a `Option` on top of that only complicates things as can be seen by the `authorities` getter method for instance.

Reply:
I think it's important to differentiate "haven't been configured" value from "configured to be empty" value. Otherwise it's very easy to cause bugs. E.g. one of such bugs I found recently with `pid` value set to be Some( 0) instead of Node, which then lead to panics

Reply:
If it won't have `Option` wrapper, in case of been empty we'll eventually forget to check for empty in some place

Reply:
I do not think there is a meaningful difference between not having an authority because none has been configured and not having an authority because it has been configured like that. Validating a credential can only succeed if there is a matching authority. In that sense it is not possible to "forget to check".

Reply:
Yes, it won't end up with accidentally verified credential in either way. But the error should be different. "Not configured authorities" vs "Trusted authority not found". The first one obviously point out what needs to be fixed, the second one makes you think that authorities were configured, but they don't match for some reason